subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -208,6 +222,7 @@ static uint32_t req_grant_delay_get(void)
 static int32_t register_callback(mpsl_cx_cb_t cb)
 {
 	callback = cb;
+	gpiote_irq_handler(NULL, NULL, 0);
```

Comment:
I don't like callbacks being called from multiple contexts. Because of this call `gpiote_irq_handler` now must be reentrant can be preempted by itself in situations like:
- low prio calls `register_callback` and gets preempted by `gpiote_irq_handler` as as result of pin toggle.
- gpiote_irq_handler is executing as a result of pin toggle end gets preempted by `register_callback`

This approach also makes it impossible to block call to `gpiote_irq_handler` if `register_callback` is called from some kind of critical section when call to `cb` is not expected to happen.

Not sure what shall we do with this.

Reply:
I think we could change 802.15.4 radio driver implementation so that it would call `mpsl_cx_granted_ops_get()` after registering the callback. This way we would not need to modify anything in mpsl_cx interface.

But then we would need to handle race conditions in the driver.

Reply:
@ankuns If I understand correctly, wherever the solution to this problem ends up, be it execution of gpiote_irq_handler in register_callback, mpsl_cx_granted_ops_get() in the driver or any other, we will always end up with possible race condition and preemption, as this problem is strictly related to data that gets changed on pin toggle and associated interrupt. 

From what I understood, the main reason for doing this in the mpsl_cx interface, as opposed to the radio driver, was unification and making the solution universal for different PHYs. 

A not-so-elegant solution to blocking the call could be a simple additional parameter in function call, or boolean "on_init" flag, the former unfortunately requiring even further modification of the interface, the latter restricting this "manual" check for initial grant state only on the first registration of callback, however I'm not sure, if this is an issue.